src/js/components/mapWidgets/widgetContent/searchContent.tsx

```diff
@@ -1,25 +1,88 @@
-import React, { FunctionComponent } from 'react';
+import React, {
+  FunctionComponent,
+  useRef,
+  useEffect,
+  useState,
+  ChangeEvent
+} from 'react';
 import { useSelector } from 'react-redux';
 
+import { mapController } from 'js/controllers/mapController';
+
+import { RootState } from 'js/store/index';
+
 import { searchContent } from '../../../../../configs/modal.config';
 
 const SearchContent: FunctionComponent = () => {
+  const searchRef = useRef(null);
+  const [formError, setFormError] = useState(false);
+  const [latitudeInput, setLatitude] = useState('');
+  const [longitudeInput, setLongitude] = useState('');
   const selectedLanguage = useSelector(
-    (state: any) => state.appState.selectedLanguage
+    (state: RootState) => state.appState.selectedLanguage
   );
-
   const { title, buttonTitle, latitude, longitude } = searchContent[
     selectedLanguage
   ];
+
+  useEffect(() => {
+    mapController.initializeSearchWidget(searchRef);
+  }, [searchRef]);
+
+  const setSearch = (): void => {
+    if (Number(latitudeInput) < 0 || Number(latitudeInput) > 90) {
+      setFormError(true);
+    } else if (Number(latitudeInput) < 0 || Number(latitudeInput) > 90) {
+      setFormError(true);
+    } else {
+      setFormError(false);
+      mapController.setSearchWidget(latitudeInput, longitudeInput);
+    }
+  };
+
```

Comment:
@SwampGuzzler here's my logic to account for latitudinal/longitudinal ranges. In pair-programming with @vaidotasp , it looks like there are instances where the latitude or longitude could be negative. I'm thinking of commenting out the conditional language for now, or checking if a number is between a negative and positive range? Feel free to let me know what you think, thanks!